crates/lovely-core/src/patch/regex.rs

```diff
@@ -94,21 +110,82 @@ impl RegexPatch {
+                        .unwrap_or_else(|| 
+                            panic!("The capture group with name '{group_name}' could not be found in '{base_str}' with the Regex '{}'", self.pattern))
+                }
+            };
+
+            let target_start = (target_group.start as isize + delta) as usize;
+            let target_end = (target_group.end as isize + delta) as usize;
+
+            let mut new_payload = String::new();
+
+            // If left border of insertion is a wordchar -> non-wordchar 
+            // boundary and our patch starts with a wordchar, prepend space so 
+            // it doesn't unintentionally concatenate with characters to its 
+            // left to create a larger identifier.
+            if self.payload.starts_with(|x: char| x.is_ascii_alphanumeric() || x == '_' || x == '$' || x == '{' || x == '}') {
+                let pre_pt = if let InsertPosition::After = self.position {
+                    target_end
+                } else {
+                    target_start
+                };
+                if pre_pt > 0 {
+                    let byte_on_left = rope.byte(pre_pt - 1);
+                    if byte_on_left.is_ascii_alphanumeric() || byte_on_left == b'_' {
+                        new_payload.push(' ');
+                    }
+                }
+            }
+
             // Prepend each line of the payload with line_prepend.
-            let new_payload = self
-                .payload
-                .lines()
-                .map(|x| if !x.is_empty() {
-                    format!("{line_prepend}{x}")
+            std::fmt::write(
+                &mut new_payload, 
+                format_args!("{}", 
+                    self
+                        .payload
+                        .split_inclusive('\n')
+                        .format_with("", |x, f| f(&format_args!("{}{}", line_prepend, x)))
+                )
+            ).expect("std::fmt::write to str should not fail");
+
+            // If right border of insertion is a non-wordchar -> wordchar 
+            // boundary and our patch ends with a wordchar, append space so 
+            // it doesn't unintentionally concatenate with characters to its 
+            // right to create a larger identifier.     
+            if self.payload.ends_with(|x: char| x.is_ascii_alphanumeric() || x == '_' || x == '$' || x == '{' || x == '}') {
+                let post_pt = if let InsertPosition::Before = self.position {
+                    target_start
                 } else {
-                    x.to_string()
-                })
-                .join("\n");
+                    target_end
+                };
+                if post_pt < rope.byte_len() {
+                    let byte_on_right = rope.byte(post_pt);
+                    if byte_on_right.is_ascii_alphanumeric() || byte_on_right == b'_' {
+                        new_payload.push(' ');
+                    }
+                }
+            }
```

Comment:
Bit of a problem with the order of operations here- usually if users use prepend_line, that adds whitespace to the left, so we should probably check for bad concatenation after prepend_line is added. Also we should probably interpolate before checking, too

Reply:
Added a check so that it doesn't add padding if the prepend_line would add that spacing.

Reply:
I'm going to put my foot down- it makes more sense, and is algorithmically simpler, to just do things in a different order: prepend_line + capture group interpolation before the concatenation checks

Reply:
In theory this removes the need to check for $ and other symbols?

Reply:
Alright, changed. And yeah, doing this check after doing all the substitutions removes the need to check for $ and other symbols.